python/cross_service/aurora_item_tracker/report.py

```diff
@@ -31,47 +42,78 @@ def __init__(self, storage, email_sender, ses_client):
+        msg = MIMEMultipart('mixed')
+        msg['Subject'] = "Work items"
+        msg['From'] = self.email_sender
+        msg['To'] = recipient
+        msg_body = MIMEMultipart('alternative')
 
-        * An HTML version that formats the report as an HTML table by using Flask's
-          template rendering feature. Email clients that can render HTML receive this
-          version.
-        * A text version that includes the report as a list of Python dicts. Email
-          clients that cannot render HTML receive this version.
+        textpart = MIMEText(text.encode(charset), 'plain', charset)
+        htmlpart = MIMEText(html.encode(charset), 'html', charset)
+        msg_body.attach(textpart)
+        msg_body.attach(htmlpart)
+
+        att = MIMEApplication(attachment.encode(charset))
+        att.add_header('Content-Disposition', 'attachment', filename='work_items.csv')
+        msg.attach(msg_body)
+        msg.attach(att)
+        return msg
+
+    @use_kwargs({'email': fields.Str(required=True)})
+    def post(self, email):
+        """
+        Gets a list of work items from storage, makes a report of them, and
+        sends an email. The email is sent in both HTMl and text format.
+
+        When ten or fewer items are in the report, the items are included in the body
+        of the email. Otherwise, the items are included as an attachment in CSV format.
 
         When your Amazon SES account is in the sandbox, both the sender and recipient
         email addresses must be registered with Amazon SES.
 
-        JSON request parameters:
-            email: The recipient's email address.
-            status: The status of work items that are included in the report.
-
-        :return: An HTTP result code.
+        :param email: The recipient's email address.
+        :return: An error message and an HTTP result code.
         """
+        response = None
         result = 200
-        parser = reqparse.RequestParser()
-        parser.add_argument('email', location='json')
-        parser.add_argument('status', location='json')
-        args = parser.parse_args()
-        work_items = self.storage.get_work_items(args['status'])
-        html_report = render_template(
-            'report.html', status=args['status'], work_items=work_items)
-        text_report = "\n".join(str(i) for i in work_items)
         try:
-            self.ses_client.send_email(
-                Source=self.email_sender,
-                Destination={'ToAddresses': [args['email']]},
-                Message={
-                    'Subject': {'Data': f"Work items: {args['status']}"},
-                    'Body': {
-                        'Html': {'Data': html_report},
-                        'Text': {'Data': text_report}}})
+            work_items = self.storage.get_work_items(archived=False)
+            snap_time = datetime.now()
+            print(f"Sending report of {len(work_items)} items to {email}.")
```

Comment:
should this use the logger, for consistency?

Reply:
fixed